### PR TITLE
Update build tools to no longer "magically" pull in related pkg dependencies (if no issues are found)

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-2111

## Summary
Removes "magically" pulling in related Bolt dependencies in our Webpack build (when unnecessary) to further help with Webpack tree shaking and reduce overall JS file size.

**Before**
image

**After**
image

^ or about `300KB` shaved off the main `bolt-global.js` bundle (even if there's just 1 component being included)!

## Details
This "feature" was originally intended to help catch and wire up undeclared dependencies within Bolt components which given how things have been trending, should no longer be required.

THAT SAID, to ensure this update doesn't accidently cause any regressions / issues downstream, the previous behavior (auto-adding missing dependencies) should continue to work if any missing packages are found + a warning gets thrown so the config can be updated (and we can get the full client-side web performance improvements of disabling this functionality).

image

## How to test
- Confirm Jest tests passing

### CODE REVIEW

File: packages/base-starter-kit/package.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
     "@bolt/components-pagination": "^2.21.0",
     "@bolt/components-placeholder": "^2.21.0",
     "@bolt/components-popover": "^2.22.0",
+    "@bolt/components-progress-bar": "^2.22.0",
```
Comment (danielamorse) [inline packages/base-starter-kit/package.json:61]: Why is this added?

File: packages/build-tools/__tests__/multi-lang/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -61,6 +61,7 @@ bolt-button[type=\\"submit\\"] {
 }
 :host {
   display: inline-flex;
+  outline: 0;
```
Comment (danielamorse) [inline packages/build-tools/__tests__/multi-lang/__snapshots__/index.js.snap:64]: Why are the snapshots changing?

**Comment by sghoweri** [comment]
> @sghoweri I might need you to walk me through what you changed. It's hard to know by looking at the diffs what changed vs what was moved, how it used to work vs how it works now. I'd like to have a better understanding before merging it. First thing tomorrow AM?

Fair enough -- walk through first thing tomorrow would be great.

File: packages/base-starter-kit/package.json (hunk 1 above)
Comment (sghoweri) [reply packages/base-starter-kit/package.json:61]: ... because it was missing?

Comment (danielamorse) [reply packages/base-starter-kit/package.json:61]: Because it was previously added magically, and it is no longer added magically?

Comment (sghoweri) [reply packages/base-starter-kit/package.json:61]: We can totally go over this tomorrow but TLDR, I don't believe this wasn't getting added even WITH magical dependencies (which still requires a dependency of a dependency to be in a package.json file somewhere).

This PR only disables magical dependencies if it doesn't find anything missing so if some other Bolt package still listed Progress Bar as a dependency then it would still get added automatically here + have a very clear message in the console warning you about it.

File: packages/build-tools/__tests__/multi-lang/__snapshots__/index.js.snap (hunk 2 above)
Comment (sghoweri) [reply packages/build-tools/__tests__/multi-lang/__snapshots__/index.js.snap:64]: The Figure and Link components aren't used in @bolt/global yet they are included in the package.json file, hence the change here: https://github.com/boltdesignsystem/bolt/pull/1864/files#diff-f35c2000901818a5ae07789735ebf5ec

**Comment by sghoweri** [comment]
@danielamorse I've nixed the `package.json` updates for `@bolt/global` which reverts the Jest snapshot update + brings down the total # of files changed here to 5:

- Updating the monorepo tests to allow `peerDependencies` to get used in our package.json files
- Adding the missing @bolt/components-progress bar that should have been in the Starter package's package.json from the getgo
- Moving the existing `aggregateBoltDependencies` function (plus supporting functions / deps) into it's own file for easier removal layer
- Moved `getPkgInfo` into its own file (to be shared across the old `aggregateBoltDependencies` function + the new logic living in `manifest.js`
- And finally updating `manifest.js`'s logic to warn about auto-added dependencies + drop adding dependencies if certain conditions are met

**Comment by sghoweri** [comment]
PR was released with v2.23.0
